client/src/Features/AssignmentLinks/useLink.hook.ts

```diff
@@ -5,15 +5,21 @@
 
 import { AssignmentLink } from '../../Models/AssignmentLink.model';
 import { useState, useEffect } from 'react';
+import { isValidURL } from './AssignmentLinkEditor';
 
 type useLinkReturnType = [AssignmentLink, (link: AssignmentLink) => void, boolean];
 
 export const useLink = (assignmentLink: AssignmentLink): useLinkReturnType => {
   const [link, setLink] = useState<AssignmentLink>(assignmentLink);
-  const [canSave, setCanSave] = useState(!!link.url);
+
+  const isValidLink = (link: AssignmentLink) => {
+    return link.url.length>0 && isValidURL(link.url) && link.displayText.length<=500 && link.description.length<=1000;

```

Comment:
Reusing the magic numbers here again is usually not scalable. We may end up updating these numbers at on place leaving the other. How about refactoring Length check function for reusability here?

Reply:
Shouldn't `isValidUrl` take care of `link.url.length > 0` check?

Reply:
For me, it would be amazing if the function could read something like below.
``` javascript
const isValidLink = ({description, displayText, url}: AssignmentLink) => isValidTitle(displayText) && isValidDesc(description) && isValidUrl(url);
```

Reply:
I have added some changes to refactor this portion.

Wrt `Shouldn't isValidUrl take care of link.url.length > 0 check?` : When the link is undefined or empty, we don't want to show the "Invalid message", but we want it to disable the `Add`/`Save` buttons. For this reason, I had added this condition here instead of inside `isValidLink`.

Reply:
Makes sense. However, in that case we should not even be calling `isUrlValid` or `isValidLink` -- right?